### DISCUSSION TITLE
different React runtime configurations result in significant differences in compiled code

### DISCUSSION THREAD

**Discussion by holynewbie** [babel/babel#15881] [Q&A] [answered]
as the code in [repl link](https://babel.dev/repl#?browsers=last%201%20Chrome%20versions&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AoUSWOAbzgCEBXGGCAOzgF85tcCAAgBMITAJ7B2mJAHoAzkhDAAtE2BlylcNHhyYYgDZI5vHHnwA6GZKFIAHhZAQhTIxblo5cjZrQc9cBwAwgbAaADWcAC8cAAUAJTRAHx05HDpcH7schBuBhAA5rH4tgBGTAUAjABM-AA0BMGhEfjxFFya5PbU8LaYKK7wsWA4YHKJUSkAPMysHHQWiyMQYzxNYeFRtOsRPGgGKF4AciggSFt6hsYW5XPsPEkAFijPj1Mys2zsSUA&debug=false&forceAllTransforms=false&modules=commonjs&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Ctypescript&prettier=true&targets=&version=7.22.10&externalPlugins=&assumptions=%7B%7D)

the **automatic**
![image](https://github.com/babel/babel/assets/8347286/58b162be-3557-4acc-b6bc-90aa9c9eb154)

the **classic**
![image](https://github.com/babel/babel/assets/8347286/deaccdda-0aac-4fc7-a3c4-41ae4de7025a)

why and how to eliminate this difference ?

**Comment by nicolo-ribaudo** [comment, marked as answer]
When using the classic runtime, you can pass the `useSpread: true` option to the JSX transform: https://babeljs.io/docs/babel-plugin-transform-react-jsx#usespread. It will default to `true` in Babel 8.

**Reply by holynewbie** [reply]
It works, tks.
